flashloan-detector/src/agent.js

```diff
@@ -20,59 +28,139 @@ const transferEventSigs = [
   "event Withdrawal(address indexed src, uint256 wad)",
 ];
 
-function provideHandleTransaction(helper, getFlashloans) {
+const transferFunctionSigs = [
+  ethers.utils.keccak256(ethers.utils.toUtf8Bytes("transfer(address,uint256)")).substring(0, 10),
+  ethers.utils.keccak256(ethers.utils.toUtf8Bytes("transferFrom(address,address,uint256)")).substring(0, 10),
+];
+
+function provideHandleTransaction(helper, getFlashloans, provider) {
   return async function handleTransaction(txEvent) {
     const findings = [];
     const initiator = txEvent.from;
 
     const flashloans = await getFlashloans(txEvent);
-    if (flashloans.length === 0) return findings;
+    const numOfFlashloans = flashloans.length;
+    if (numOfFlashloans === 0) return findings;
 
+    const calledContract = txEvent.to.toLowerCase();
     const transferEvents = txEvent.filterLog(transferEventSigs);
     const { traces } = txEvent;
 
+    let totalTokenProfits = {};
+    let totalNativeProfit = helper.zero;
+    let totalBorrowed = 0;
+
     // For each flashloan calculate the token profits and the borrowed amount
-    const flashloansData = await Promise.all(
-      flashloans.map(async (flashloan) => {
+    await Promise.all(
+      flashloans.map(async (flashloan, flashloanIndex) => {
         const { asset, amount, account } = flashloan;
 
-        let tokenProfits = {};
-        let nativeProfit = helper.zero;
-
         if (account !== initiator) {
-          tokenProfits = helper.calculateTokenProfits(transferEvents, account);
-          nativeProfit = helper.calculateNativeProfit(traces, account);
+          const tokenProfits = helper.calculateTokenProfits(transferEvents, account);
+          const nativeProfit = helper.calculateNativeProfit(traces, account);
+
+          Object.entries(tokenProfits).forEach(([address, profit]) => {
+            if (!totalTokenProfits[address]) totalTokenProfits[address] = helper.zero;
+            totalTokenProfits[address] = totalTokenProfits[address].add(profit);
+          });
+          totalNativeProfit = nativeProfit.add(nativeProfit);
```

Comment:
```suggestion
          totalNativeProfit = totalNativeProfit.add(nativeProfit);
```

Should this be like the change?